# /qompassai/linux/patches/sgx-main-get_unmapped_area-fix.patch
# Fix SGX driver build for Linux Kernel 6.14.9 Nitrous
# ------------------------------------------------------------------

--- sgx_main.c
+++ sgx_main.c
@@ -140,7 +140,12 @@
 	unsigned long addr, vm_flags;

 	down_write(&current->mm->mmap_sem);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,14,0)
+	addr = vm_unmapped_area(&info);
+#else
 	addr = current->mm->get_unmapped_area(file, addr, 2 * len, pgoff,
+#endif
 		flags);
 	up_write(&current->mm->mmap_sem);

@@ -387,7 +392,7 @@
  * Module initialization
  * Note: This must be called after the enclave device is initialized!
  */
-int init_sgx_module(void)
+static int __init init_sgx_module(void)
 {
 	int ret;

@@ -396,7 +401,7 @@
 	return ret;
 }

-void cleanup_sgx_module(void)
+static void __exit cleanup_sgx_module(void)
 {
 	driver_remove_file(&sgx_drv.driver, &driver_attr_enclave);
 	driver_unregister(&sgx_drv);

